@mui/lab has incorrect peer dependency version of @mui/material

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

1. create a project using mui lab TabPanel component (latest version of @mui/lab)
2. install any @mui/material version below 5.10.11
3. run project
4. observe error message `Module not found: Error: Can't resolve '@mui/material/generateUtilityClasses' in '......\@mui\lab\TabPanel'`

### Current behavior 😯

An error occurs

### Expected behavior 🤔

No error.

The peer dependency on @mui/material should be specified as ^5.10.11 because that is the minimum version that works 
https://github.com/mui/material-ui/releases/tag/v5.10.11 - the required changes were part of this pull request https://github.com/mui/material-ui/pull/33411

### Context 🔦

_No response_

### Your environment 🌎

_No response_

--- a/packages/mui-lab/package.json
+++ b/packages/mui-lab/package.json
@@ -65,7 +65,7 @@
   "peerDependencies": {
     "@emotion/react": "^11.5.0",
     "@emotion/styled": "^11.3.0",
-    "@mui/material": "^5.0.0",
+    "@mui/material": ">=5.10.11",
     "@types/react": "^17.0.0 || ^18.0.0",
     "react": "^17.0.0 || ^18.0.0",
     "react-dom": "^17.0.0 || ^18.0.0"

